code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -67,18 +67,17 @@
 		var/turf/T = get_step(src, direction)
 		if(!istype(T))
 			continue
-		var/counterdir = get_dir(T, src)
 		if(CanAtmosPass(T))
 			atmos_adjacent_turfs_amount += 1
-			atmos_adjacent_turfs |= direction
-			if(!(T.atmos_adjacent_turfs & counterdir))
+			atmos_adjacent_turfs |= T
+			if(!(src in T.atmos_adjacent_turfs))

```

Comment:
Isn't the 'in' keyword very slow? [According to this (though I think there is a typo with it saying 'not' instead of 'in) it is slow.](https://gist.github.com/Giacom/be635398926bb463b42a#file-byond-performance-txt-L434)

Maybe using an associated list (which I'm not sure is a hashmap or not) could be faster.

I would try testing it first.